### ISSUE TITLE
Better decorations off task note links

### ISSUE THREAD

**Issue by jokaro** [dendronhq/dendron#2955]
## Please select if your request is either something new or an enhancement ##

- [x] Enhancement of an existing Feature.
- [ ] Request of a new feature.

## Please select the area your request applies to. (Multiple selections are Possible. You can leave blank if you're not sure.) ##

- [ ] Workspace - VSCode workspace, vaults, Intellisense/autocomplete, Dendron settings
- [ ] Lookup - Dendron's Lookup Command
- [ ] Views - Dendron Preview, Tree View, Side Panels in the UI
- [ ] Schema - Dendron Schemas
- [ ] Pod - Data import from / export to Dendron
- [ ] Publish - External Site Publish
- [x] Markdown - Markdown features, such as syntax support and features

## Is your feature request related to a problem? Please describe ##

A list of linked task notes makes it hard to parse the information since all text have the same color.

![image](https://user-images.githubusercontent.com/210008/169809036-9367ad17-97a5-4d40-b6aa-20e8a9d3dd07.png)

## Describe the solution you'd like ##

It would be better if the decorated information would stand out from the link with a separate color. Like this screenshot I found in the dendron wiki.

![image](https://user-images.githubusercontent.com/210008/169807996-ef0aaab6-587c-4cbd-aac9-4341a834665e.png)
https://wiki.dendron.so/notes/SEASewZSteDK7ry1AshNG/#creating-a-task-note

So either this is a mock image or it was like this before. 

## Describe alternatives you've considered ##

- It would be even better if this would be configurable, setting font, size and color of each element separately. 
- Adding the colored tag square would also streamline this with the regular task lists

## Additional context ##

None

**Comment by Harshita-mindfire** [comment]
Thanks @jokaro for your suggestions 🙏

**Comment by SeriousBug** [comment]
@jokaro You are right that it used to be regular text color, but it would blend in with the regular text around the link then. 

For whoever works on this issue, you'll want to pick a color from here so it follows the users current theme: https://code.visualstudio.com/api/references/theme-color

**Comment by jokaro** [comment]
Maybe try out if using inlay hints is a better approach
![image](https://user-images.githubusercontent.com/210008/169982850-105be4f4-310a-46d7-91d7-14c9db393bd4.png)

I don't know how they work or if they have any limitations. But then it wouldn't blend with the regular text.
